core/src/main/java/ar/com/wolox/wolmo/core/service/WoloxRetrofitServices.java

```diff
@@ -15,43 +15,46 @@
 public abstract class WoloxRetrofitServices {
 
     private Retrofit mRetrofit;
-    private Retrofit mApiaryRetrofit;
     private Map<Class, Object> mServices;
 
     public void init() {
         mServices = new HashMap<>();
-        Gson gson = GsonBuilder.getBasicGsonBuilder().create();
-
-        HttpLoggingInterceptor loggerInterceptor = new HttpLoggingInterceptor();
-        loggerInterceptor.setLevel(HttpLoggingInterceptor.Level.BODY);
-
-        OkHttpClient client = new OkHttpClient.Builder()
-                .addInterceptor(new SecuredRequestInterceptor())
-                .addInterceptor(loggerInterceptor)
-                .build();
-
         mRetrofit = new Retrofit.Builder()
                 .baseUrl(getApiEndpoint())
-                .addConverterFactory(GsonConverterFactory.create(gson))
-                .client(client)
-                .build();
-        mApiaryRetrofit = new Retrofit.Builder()
-                .baseUrl(getApiaryEndpoint())
-                .addConverterFactory(GsonConverterFactory.create(gson))
-                .client(client)
+                .addConverterFactory(GsonConverterFactory.create(getGson()))
+                .client(getOkHttpClient())
                 .build();
     }
 
+    /**
+     * Returns the API endpoint.
+     *
+     * @return URL endpoint
+     */
     public abstract String getApiEndpoint();
 
-    public abstract String getApiaryEndpoint();
+    /**
+     * Returns an instance of Gson to use for conversion.
+     *
+     * @return new Gson instance
+     */
+    protected Gson getGson() {
+        return GsonBuilder.getBasicGsonBuilder().create();
+    }
 
-    public <T> T getApiaryService(Class<T> clazz) {
-        T service = (T) mServices.get(clazz);
-        if (service != null) return service;
-        service = mApiaryRetrofit.create(clazz);
-        mServices.put(clazz, service);
-        return service;
+    /**
+     * Build and configure a new instance of OkHttpClient.
+     * You must add interceptors and configure the client before returning it.
+     *
+     * @return A configured instance of OkHttpClient
+     */
+    protected OkHttpClient getOkHttpClient() {
+        HttpLoggingInterceptor loggerInterceptor = new HttpLoggingInterceptor();
+        loggerInterceptor.setLevel(HttpLoggingInterceptor.Level.BODY);
+
+        return new OkHttpClient.Builder()
+                .addInterceptor(loggerInterceptor)
```

Comment:
I did not realize we could make `getOkHttpClient` protected so if the subclass wants a custom implementation they can override and change it.
I like the way you say, because the client does not need to worry about the creation if it does not need a custom implementation (like most of the time).
I'll update te code :smile: